src/txn/txn.c

```diff
@@ -441,20 +441,61 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
+		char timestamp_buf[2 * WT_TIMESTAMP_SIZE + 1];
+		bool round_to_oldest;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
-		__wt_timestamp_set(&txn->read_timestamp, &ts);
+		WT_RET(__wt_config_gets_def(session, cfg, "round_to_oldest", 0,
+		    &cval));
+		round_to_oldest = false;
+		__wt_readlock(session, &txn_global->rwlock);
+		if (__wt_timestamp_cmp(&ts, &txn_global->oldest_timestamp) < 0)
+		{
+			/*
+			 * To circumvent error: unused-result
+			 */
+			(void)(__wt_timestamp_to_hex_string(session,
+			    timestamp_buf, &ts) + 1);
+			round_to_oldest = cval.val;
```

Comment:
Why not move this outside the if block, where you parse the value passed for round_to_oldest. We are initializing to false there, why not just set to the value user passed there itself.

Reply:
because this block is not required, when if is evaluated to false.

Reply:
That looks a bit inconsistent to me. Either of the following two look consistent:

1. parse value for "round_to_oldest" config parameter outside the if block and set local variable round_to_oldest to that value there itself.
2. set round_to_oldest to false outside the if block and then inside if block parse value for "round_to_oldest" config parameter and set round_to_oldest.

I like option 1 better, but that is my opinion. The code currently mixes the above two approaches.
I would let second reviewer look at this, as this is more of code layout than a correctness thing